tests/getmempolicy/getmempolicy.c

```diff
@@ -0,0 +1,191 @@
+#define MPOL_MF_MOVE_ALL (1 << 2)
+
+long get_mempolicy(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    return syscall(SYS_get_mempolicy, mode, nodemask, maxnode, addr, flags);
+}
+
+void print(
+    int* mode,
+    unsigned long* nodemask,
+    unsigned long maxnode,
+    void* addr,
+    unsigned long flags)
+{
+    printf(
+        "\nmode = %d\tnodemask = %ld\tmaxnode = %ld\taddr = %p\tflags = %lu",
+        *mode,
+        *nodemask,
+        maxnode,
+        addr,
+        flags);
+}
+
+int test1()
+{
+    int mode = 0xffffffff;
+    unsigned long nodemask = 0xffffffffffffffff;
+    unsigned long maxnode = 64;
+    void* addr = NULL;
+    unsigned long flags = MPOL_DEFAULT;
+
+    printf("1. MPOL_DEFAULT, null addr");
+    print(&mode, &nodemask, maxnode, addr, flags);
+    long ret = get_mempolicy(&mode, &nodemask, maxnode, addr, flags);
+    print(&mode, &nodemask, maxnode, addr, flags);
+    printf("\nRet = %ld", ret);
+    printf("\nerrno = %d", errno);
+
+    assert(ret == 0);
+    assert(mode == 0);
+    assert(nodemask == 0);
+    assert(maxnode == 64);
+    assert(addr == 0);
+    assert(flags == MPOL_DEFAULT);
+}
+
+int test2()
+{
+    int mode = 0xffffffff;
+    unsigned long nodemask = 0xffffffffffffffff;
+    unsigned long maxnode = 64;
+    void* addr = NULL;
+    unsigned long flags = MPOL_F_MEMS_ALLOWED;
+
+    printf("\n\n2. MPOL_F_MEMS_ALLOWED, null addr");
+    print(&mode, &nodemask, maxnode, addr, flags);
+    long ret = get_mempolicy(&mode, &nodemask, maxnode, addr, flags);
+    print(&mode, &nodemask, maxnode, addr, flags);
+    printf("\nRet = %ld", ret);
+    printf("\nerrno = %d", errno);
+
+    assert(ret == 0);
+    assert(mode == 0);
+    assert(nodemask == 0); // mystikos=0 native=1 since mystikos does not
+    // support mbind and set_mempolicy
+    assert(maxnode == 64);
+    assert(addr == 0);
+    assert(flags == MPOL_F_MEMS_ALLOWED);
+}
+
+int test3()
+{
+    int mode = 0xffffffff;
+    unsigned long nodemask[5];
+    unsigned long maxnode = 41;
+    void* addr = NULL;
+    unsigned long flags = MPOL_DEFAULT;
+
+    printf("\n\n3. MPOL_DEFAULT, maxnode > sizeof(nodemask)*BIT_TO_BYTE");
+    print(&mode, (unsigned long*)nodemask, maxnode, addr, flags);
+    long ret =
+        get_mempolicy(&mode, (unsigned long*)nodemask, maxnode, addr, flags);
+    print(&mode, (unsigned long*)nodemask, maxnode, addr, flags);
+    printf("\nRet = %ld", ret);
+    printf("\nerrno = %d", errno);
+    assert(ret == 0);
+    assert(errno == 0);
+    assert(mode == 0);
+    assert(nodemask[0] == 0);
+    assert(maxnode == 41);
+    assert(addr == 0);
+    assert(flags == MPOL_DEFAULT);
+}
+
+int test4()
+{
+    int mode = 5;
+    unsigned long nodemask = 0xffffffffffffffff;
+    unsigned long maxnode = 64;
+    void* addr = NULL;
+    unsigned long flags = MPOL_F_ADDR;
+
+    printf("\n\n4. MPOL_F_ADDR, NULL addr, expecting EFAULT");
```

Comment:
Should also test valid addr case. If addr points to a buffer allocated on heap or stack, it's part of the Mystikos app EPC.